ci/run_ci.sh

```diff
@@ -28,38 +30,32 @@ diff_list=`git diff origin/master --name-only`
 # - anything in ci has been changed
 # - anything in utils has been changed
 # Else only run tests on directories that have changed
-if [[ `echo $diff_list | grep -v / | wc -l` -gt 0 || `echo $diff_list | grep ci/` || `echo $diff_list | grep utils/` ]]
-then
+if [[ `echo ${diff_list} | grep -cv /` -gt 0 || `echo ${diff_list} | grep -E "(ci|utils|image_resources)/"` ]]; then
   echo "Running full test"
-  test_list=`ls -d */ | grep -Ev "(utils|ci|ripsaw|image_resources)/"`
+  test_list=`find * -name ci_test.sh -type f -exec dirname {} \;`
 else
   echo "Running specific tests"
   echo $diff_list
-  test_list=`echo $diff_list | awk -F "/" '{print $1}' | uniq`
+  test_list=`echo "${diff_list}" | xargs dirname | uniq`
```

Comment:
call me paranoid, but if someone creates a wrapper with subdirectories, would this work?   I think the awk command would.    I don't think any existing wrappers have subdirectories, but that could change, there is nothing requiring a wrapper not to use subdirectories.   I don't think this is an essential change, do you?

Reply:
You're right, this change is not essential. It is just an attempt to get the rid of awk here, which sometimes can be confusing, as this command is not focused on path parsing but in strings processing. You're also right in the fact that we're not currently using sub-directories, dirname might be problematic here.